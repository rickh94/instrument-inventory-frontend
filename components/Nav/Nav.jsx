import React, { useState } from 'react'
import { withStyles } from '@material-ui/core/styles'
import PropTypes from 'prop-types'
import { Link as RouterLink } from 'react-router-dom'

import MenuIcon from '@material-ui/icons/Menu'
import PersonIcon from '@material-ui/icons/Person'
import {
  IconButton,
  Toolbar,
  AppBar,
  Typography,
  Button,
  SwipeableDrawer,
  List,
  ListItem,
  ListItemText,
  Divider,
  ListItemIcon
} from '@material-ui/core'

const styles = {
  root: {
    flexGrow: 1
  },
  grow: {
    flexGrow: 1
  },
  barText: {
    color: 'white'
  },
  cleanLink: {
    textDecoration: 'none',
    color: 'inherit'
  }
}

export function Nav(props) {
  const { classes } = props
  const [drawerIsOpen, setDrawerOpen] = useState(false)
  return (
    <React.Fragment>
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <IconButton
              color="inherit"
              aria-label="Menu"
              onClick={() => setDrawerOpen(true)}
            >
              <MenuIcon />
            </IconButton>
            <Typography variant="h6" color="inherit" className={classes.grow}>
              Instrument Inventory
            </Typography>
            {props.isAuthenticated ? (
              <Button color="inherit" onClick={props.handleLogout} id="#logout">
                Logout
              </Button>
            ) : (
              <RouterLink to="/login">
                <Button color="inherit" className={classes.barText}>
                  Login
                </Button>
              </RouterLink>
            )}
          </Toolbar>
        </AppBar>
      </div>
      <SwipeableDrawer
        open={drawerIsOpen}
        onClose={() => setDrawerOpen(false)}
        onOpen={() => setDrawerOpen(false)}
      >
        <List>
          <ListItem button>
            <ListItemText>
              <RouterLink to="/" className={classes.cleanLink}>Home</RouterLink>
            </ListItemText>
          </ListItem>
          <Divider></Divider>
          <ListItem button>
          <ListItemIcon><PersonIcon /></ListItemIcon>
            <ListItemText>
              <RouterLink to="/profile" className={classes.cleanLink}>Profile</RouterLink>
            </ListItemText>
          </ListItem>
        </List>
      </SwipeableDrawer>
    </React.Fragment>
  )
}

Nav.propTypes = {
  classes: PropTypes.object.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  handleLogout: PropTypes.func.isRequired
}

export default withStyles(styles)(Nav)
