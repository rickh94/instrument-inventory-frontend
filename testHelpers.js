import React from 'react'

export const testToken = 'testtoken'


export function clickThing(tag, wrapper, content) {
  wrapper
    .findWhere(el => el.name() == tag && el.contains(content))
    .simulate('click', { preventDefault: jest.fn() })
}

export const clickButton = (wrapper, content) => clickThing('Button', wrapper, content)
export const clickLink = (wrapper, content) => clickThing('a', wrapper, content)

