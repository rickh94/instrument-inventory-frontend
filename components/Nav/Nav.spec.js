import React from 'react'
import { shallow } from 'enzyme'
import { Nav } from './Nav'

import '../../setupTests'
import { clickButton } from '../../testHelpers'

describe('<Nav />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Nav
        isAuthenticated={false}
        handleLogout={jest.fn()}
        classes={{ root: 'root', grow: 'grow', barText: 'barText' }}
      />
    )
  })

  test('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('shows login if logged out', () => {
    expect(wrapper.contains('Login')).toBeTruthy()
  })

  it('shows logout if logged in', () => {
    wrapper.setProps({ isAuthenticated: true })
    expect(wrapper.contains('Logout')).toBeTruthy()
  })

  it('calls function if logout clicked', () => {
    const handleLogout = jest.fn()
    wrapper.setProps({ isAuthenticated: true, handleLogout })
    // console.log(wrapper.findWhere(el => el.contains('Logout')))
    // expect(handleLogout).toHaveBeenCalled()
  })
})
