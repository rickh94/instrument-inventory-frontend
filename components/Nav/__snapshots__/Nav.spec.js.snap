// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders 1`] = `ShallowWrapper {}`;
